src/axolotl/train.py

```diff
@@ -134,6 +134,22 @@ def terminate_handler(_, __, model):
     # only save on rank 0, otherwise it corrupts output on multi-GPU when multiple processes attempt to write the same file
     if cfg.fsdp:
         trainer.save_model(cfg.output_dir)
+    elif cfg.deepspeed:
```

Comment:
does this apply to all zero* stages, or just zero3 as listed in the pr title?

Reply:
this error applies only for zero3. this code change will run for all zero* stages. should work for all zero* stages however.

if we want a smaller change, i think something like this should work:
https://github.com/lm-sys/FastChat/pull/1457/files#diff-82b734e9eda6b4bac9a28b1056d4e0e0676f904e43cede16d7aa6e2d1da3e61bR155

Reply:
perhaps `elif cfg.deepspeed and trainer.hf_deepspeed_config_orig.is_zero3():` just to minimize the blast radius on this change?